requests.go

```diff
@@ -42,18 +41,18 @@ type userData struct {
 	SessionToken string `json:"sessionToken"`
 }
 
-func getCookie(c *http.Client, urlstr, cookieName string) *http.Cookie {
+func getCookie(c *http.Client, urlstr, cookieName string) (*http.Cookie, error) {
 	urlObj, err := url.Parse(urlstr)
 	if err != nil {
 		panic(err)
 	}
 	cookies := c.Jar.Cookies(urlObj)
 	for _, c := range cookies {
 		if c.Name == cookieName {
-			return c
+			return c, nil
 		}
 	}
-	return nil
+	return nil, err
```

Comment:
are you meaning to return an error if there is no cookie? here you still return `nil` error so I think the code will still panic below

Reply:
This is updated in commit bd12eec453bd4a9.
`return nil, fmt.Errorf("failed to find %s in cookie jar", cookieName)`